### PR TITLE
chore: update MDC to latest canary

### PR DESCRIPTION
**Pull request by mmalerba**

### CODE REVIEW

File: src/material-experimental/mdc-chips/chip-option.ts [hunk 1]
```diff
@@ -200,10 +200,11 @@ export class MatChipOption extends MatChip {
 
   /** Handles click events on the chip. */
   _click(event: MouseEvent) {
+    console.error('CHIP OPTION CLICK');
```
Comment (devversion) [inline src/material-experimental/mdc-chips/chip-option.ts]: Leftover from debugging.

File: src/material-experimental/mdc-chips/chip-icons.ts [hunk 2]
```diff
@@ -106,16 +102,22 @@ const _MatChipRemoveMixinBase:
...
+    '(keydown)': 'keydowns.next($event)',
   }
 })
 export class MatChipRemove extends _MatChipRemoveMixinBase implements CanDisable, HasTabIndex {
   /**
-   * Emits when the user interacts with the icon.
+   * Emits when the user clicks the icon.
+   * @docs-private
+   */
+  clicks: Subject<MouseEvent> = new Subject<MouseEvent>();
```
Comment (crisbeto) [inline src/material-experimental/mdc-chips/chip-icons.ts]: Should we underscore this if it's not supposed to be public?

File: src/material-experimental/mdc-chips/chip.ts [hunk 3]
```diff
@@ -321,24 +321,31 @@ export class MatChip extends _MatChipMixinBase implements AfterContentInit, Afte
...
   /** Handles interaction with the remove icon. */
-  _listenToRemoveIconInteraction() {
-    this.removeIcon.interaction
+  _listenToRemoveIconInteractions() {
+    this.removeIcon.clicks
         .pipe(takeUntil(this._destroyed))
-        .subscribe((event) => {
+        .subscribe(event => {
+          if (this.disabled) {
+            return;
```
Comment (crisbeto) [inline src/material-experimental/mdc-chips/chip.ts]: The early return here seems weird considering that there's only one statement afterwards. Maybe have it as `if (!this.disabled) {...}`.

File: src/material-experimental/mdc-chips/chip-icons.ts [hunk 4]
```diff
@@ -106,16 +102,22 @@ const _MatChipRemoveMixinBase:
...
+
+  /**
+   * Emits when the user presses a key on the icon.
    * @docs-private
    */
-  interaction: Subject<MouseEvent | KeyboardEvent> = new Subject<MouseEvent | KeyboardEvent>();
+  keydowns: Subject<KeyboardEvent> = new Subject<KeyboardEvent>();
```
Comment (crisbeto) [inline src/material-experimental/mdc-chips/chip-icons.ts]: These subjects aren't being completed on destroy as far as I can tell.

**Pull request merged**